docs/interaction_based_testing.adoc

```diff
@@ -1056,6 +1056,30 @@ mock.type == List
 mock.nature == MockNature.MOCK
 ----
 
+[[DetachedMockFactory]]
+=== Create Mock Objects outside Specifications
+
+It can be sometimes helpful to create mock, stub or spy objects outside a `Specification`, but use them in a `Specification`.
+The `spock.mock.DetachedMockFactory` provides API to create such mock objects.
+
+NOTE: Mocks created by the `DetachedMockFactory` must be manually attached to the `Specification`
```

Comment:
I think we need another note, which explains that the mocks can't have any behavior unless they are attached to a spec. Or, you use a [custom IResponseGenerator](https://stackoverflow.com/a/74804706/2145769), but I'm not sure if we should include the last part as an example.

Reply:
Done.